### ISSUE TITLE
Improve importer performance.

### ISSUE THREAD

**Issue by oliverchang** [google/osv.dev#576]

**Comment by another-rex** [comment]
> GHSA repo can be slow to clone. Maybe we can download a tarball instead?

Can also be sped up with a shallow clone. (5m48s for full clone locally, down to 37s if doing a clone for only 1 week of commits). 

The current import code requires git history because it walks through commits to get the differences between last update and the new update. This can probably be updated to manually diff without git, but it's going to be easier to implement by just using `--shallow-since` and input the previous update date.

**Comment by oliverchang** [comment]
> > GHSA repo can be slow to clone. Maybe we can download a tarball instead?
> 
> Can also be sped up with a shallow clone. (5m48s for full clone locally, down to 37s if doing a clone for only 1 week of commits).
> 
> The current import code requires git history because it walks through commits to get the differences between last update and the new update. This can probably be updated to manually diff without git, but it's going to be easier to implement by just using `--shallow-since` and input the previous update date.

Right, we can likely skip this if we can just rely solely on `last_modified`. We can do a shallow clone with --depth=1 in that case. But checking against `last_modified` may still be slow in its current form unless we batch these requests.

**Comment by another-rex** [comment]
Tested git pull on a shallow repository, and it does only pull the changes, without needing to fill out the full history.
